Heisenberg4269/LCA.cpp

```diff
@@ -0,0 +1,83 @@
+class Solution
+{
+public:
+    vector < vector<int> > tree;
+    vector <int> depth;
+    vector < vector<int> > ancestor;
+
+
+    Solution(vector < vector<int> > fn_tree)
```

Comment:
`Solution(vector <vecot<int>>& fn_tree)`. Pass by reference works faster. In cases where you are passing an object of a class as a parameter if it is not passed by reference then copy constructor of that class would be called.

Reply:
`const vector<vector<int>> &fn_tree`, if you are accepting something by reference and don't intend on modifying its contents, the parameter should be a const. This comment applies throughout this PR.